SIG Network Annual Report 2024 (#8356)
kubernetes/community: sig-network/annual-report-2024.md

1. What work did the SIG do this year that should be highlighted?

This year we have several highlights from our various sub-projects, these are
the major themes and releases:

- [External DNS](https://github.com/kubernetes-sigs/external-dns)
  - Two new reviewers were added this year as the project continues to grow
  - Shipped important maintenance release [v0.15.x](https://github.com/kubernetes-sigs/external-dns/releases/tag/v0.15.0)

> - [Ingress Nginx](https://github.com/kubernetes/ingress-nginx)

Reviewer: "Ingress NGINX"

Author: :+1: fixed

  - Shipped a large number of `v1.x.x` patch and minor releases all the way from `v1.8.x` to [v1.12.0](https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v1.12.0)

> - [Ingate](https://github.com/kubernetes-sigs/ingate)

Reviewer: "InGate"

Author: :+1: fixed

  - Project just started at the end of 2024 and is starting to plan the roadmap.

> - [MultiNetwork](https://github.com/kubernetes-sigs/multi-network-api)

Reviewer: "Multi-Network"? Also, that repo is still just an empty template repo.

Author: :+1: fixed

And yes, that's why I'm calling it out below as needing help.

  - Project just started at the end of 2024 and is starting to plan the roadmap.
- [Multi-Network](https://github.com/kubernetes-sigs/multi-network-api)
  - Moved away from core development to CRD based addon approach
  - Work started on supporting [Dynamic Resource Allocation (DRA)]
- [Network Policy](https://github.com/kubernetes-sigs/network-policy-api)
  - Working on getting `AdminNetworkPolicy` (ANP) and `BaselineAdminNetworkPolicy` (BANP) APIs to `Beta`
    - Two new implementations joined us this year in supporting the APIs
  - Delivered features: [FQDN Selectors], [Policy Assistant CLI], and [Tenancy API]
  - Work started on [Service Account Selectors], and [Dry-Run Mode]
- [Gateway API](https://github.com/kubernetes-sigs/gateway-api)
  - Delivered [GRPCRoute], [ParentReference Port], [Service Mesh Support] and [Conformance Profiles and Reports] as GA in release [v1.1][gwv1.1]
  - Delivered [HTTPRoute Timeouts], [Gateway Infrastructure Labels], and [Backend Protocol Support] as GA in release [v1.2][gwv1.2]
  - Shipped two releases of our [ingress2gateway] utility: [v0.2.0][i2gv0.2], [v0.3.0][i2gv0.3]
  - Shipped our first release of our [gwctl] utility: [v0.1.0][gwctlv0.1]

> - [Gateway API Inference Extensions (GIE)](https://github.com/kubernetes-sigs/gateway-api-inference-extension)
>   - Shipped our first release [v0.1.0](https://github.com/kubernetes-sigs/gateway-api-inference-extension)

Reviewer 2: ```suggestion
- [Gateway API Inference Extensions (GIE)](https://github.com/kubernetes-sigs/gateway-api-inference-extension)
  - Shipped our first release [v0.1.0](https://github.com/kubernetes-sigs/gateway-api-inference-extension)
  - Reduce  output token latency by [15%-60%] when kv cache is close to saturation
  - Current integration efforts with the following gateways:
    - GKE
    - [KGateway](https://kgateway.dev/) 
    - [Istio](https://istio.io/)
  - [Try it out](https://gateway-api-inference-extension.sigs.k8s.io/guides/) or take a look at our [roadmap](https://github.com/kubernetes-sigs/gateway-api-inference-extension?tab=readme-ov-file#roadmap) to see whats coming up next!
  
```

Author: Thanks @kfswain!

Just a note that I'll keep most of this, but I'll drop the "Try it out" since this section about covering our major achievements, releases, and such.

If however you want to take a look at my "call for help" for the project below, and maybe suggest a change to that to be more clear that you're looking for feedback etc that would be a reasonable place to add that :+1:

1. What work did the SIG do this year that should be highlighted?

This year we have several highlights from our various sub-projects, these are
the major themes and releases:

- [External DNS](https://github.com/kubernetes-sigs/external-dns)
  - Two new reviewers were added this year as the project continues to grow

>   - Shipped important maintenance release [v0.15.x](https://github.com/kubernetes-sigs/external-dns/releases/tag/v0.15.0)

Reviewer 3: It would be great to add to external-dns:
```
   - Created [webhook provider](https://kubernetes-sigs.github.io/external-dns/v0.14.2/tutorials/webhook-provider/) to enable providers running out of tree. Our goal is to move all providers out of tree.
```

Author: Sure thing!

```suggestion
  - Shipped important maintenance release [v0.15.x](https://github.com/kubernetes-sigs/external-dns/releases/tag/v0.15.0)
  - Created [webhook provider](https://kubernetes-sigs.github.io/external-dns/v0.14.2/tutorials/webhook-provider/) to enable providers running out of tree. Our goal is to move all providers out of tree.
```

2. Are there any areas and/or subprojects that your group needs help with (e.g. fewer than 2 active OWNERS)?

While there are no SIG Network projects which _couldn't_ benefit from more help
and contributions, the following is a list of specific projects where there are
known blockers (or otherwise critical needs) where more contributions, feedback,
or implementation support is the key to "unblocking" them:

- [InGate] is a new project just getting started creating a Gateway API implementation, and needs contributors to join and get to the first release.
- [Network Policy] could use more implementations to join and support some of our newer APIs (e.g. [AdminNetworkPolicy]) to help them graduate towards GA.
- [MultiNetwork] has started anew with CRD based development and is in need of use cases and implementations to join to get to the first release.
- [Gateway API]
  - has a large number of [features which are not yet standard] which could use implementations to join us to help us graduate those towards GA.
  - has two CLI utilities which need more users and implementations to get involved to provide feedback: [ingress2gateway], [gwctl]

> - [Gateway API Inference Extension] is a new project which provides AI/ML inference routing and load-balancing features. It is in need of more implementations to bring use cases and and contribute in order to help graudeate the APIs towards GA.

Reviewer 2: ```suggestion
- [Gateway API Inference Extension] is a new project which provides AI/ML inference routing and load-balancing features. It is in need of more implementations to bring use cases and and contribute in order to help graduate the APIs towards GA.
  - Additionally, user feedback is needed, and _always_ appreciated, so please [try it out](https://gateway-api-inference-extension.sigs.k8s.io/guides/)!
```

Author: ty! fixed

- Stable
  - [1860 - Make Kubernetes aware of the load balancer behaviour](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1860-kube-proxy-IP-node-binding) - v1.32
  - [2433 - Topology Aware Hints](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2433-topology-aware-hints) - v1.30
  - [2681 - Field status.hostIPs added for Pod](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2681-pod-host-ip) - v1.30
  - [3458 - Remove transient node predicates from KCCM's service controller](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3458-remove-transient-node-predicates-from-service-controller) - v1.30
  - [3705 - Cloud Dual-Stack --node-ip Handling](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3705-cloud-node-ips) - v1.30
  - [3836 - Kube-proxy improved ingress connectivity reliability](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3836-kube-proxy-improved-ingress-connectivity-reliability) - v1.31

> 

Reviewer: Did you double-check these? Last year the script that generated this section wasn't totally reliable...

Author: Double checked:

- 4427, 1880, 3866, 4444, 1860, 2681, 3458, 3705  :heavy_check_mark: 
- 784, 2433 it was wrong about, fixed :+1:

3. Did you have community-wide updates in 2024 (e.g. KubeCon talks)?

Blog Posts:

- [Getting started with the AdminNetworkPolicy](https://network-policy-api.sigs.k8s.io/blog/2024/01/30/getting-started-with-the-adminnetworkpolicy-api/)
- [Gateway API v1.1](https://kubernetes.io/blog/2024/05/09/gateway-api-v1-1/)
- [Gateway API v1.2](https://kubernetes.io/blog/2024/11/21/gateway-api-v1-2/)

Talks:

- [Network Policy: The future of Network Policy (Kubecon EU)](https://sched.co/1Yhhb)
- [How many network policies can I create? (Kubecon EU)](https://kccnceu2024.sched.com/event/1YeMI)

> - [Gateway API "Beyond GA" (Kubecon EU)](https://www.youtube.com/watch?v=LITg6TvctjM)

Reviewer: Should probably be consistent about sched vs youtube links?
Or have both for all of them (since you can usually get the slides at the sched link).

Author: Yeah, fixed :+1:
